src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/DraftOrderService.java

```diff
@@ -131,4 +141,98 @@ private AgreedDraftOrder mapToAgreedDraftOrderForPsa(
         builder.orderStatus(OrderStatus.TO_BE_REVIEWED);
         return applySubmittedInfo(userAuthorisation, builder.build());
     }
+
+    public void populateDraftOrdersReviewCollection(FinremCaseData finremCaseData, UploadAgreedDraftOrder uploadAgreedDraftOrder,
+                                                    List<AgreedDraftOrderCollection> newAgreedDraftOrderCollection) {
+        // ensure non-null hearing details
+        if (uploadAgreedDraftOrder.getHearingDetails() == null) {
+            log.error("Unexpected null hearing details for Case ID: {}", finremCaseData.getCcdCaseId());
+            return;
```

Comment:
Would it be better to throw an exception if we don't have either the hearing details or judge?